pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -77,25 +79,27 @@ type DiscoveryServer struct {
 }
 
 // NewDiscoveryServer creates DiscoveryServer that sources data from Pilot's internal mesh data structures
-func NewDiscoveryServer(env model.Environment, generator core.ConfigGenerator) *DiscoveryServer {
+func NewDiscoveryServer(env *model.Environment, generator core.ConfigGenerator) *DiscoveryServer {
 	out := &DiscoveryServer{
 		env:             env,
 		ConfigGenerator: generator,
 	}
+	env.PushStatus = model.NewStatus()
 
 	envOverride := os.Getenv("V2_REFRESH")
 	if len(envOverride) > 0 {
 		var err error
 		periodicRefreshDuration, err = time.ParseDuration(envOverride)
 		if err != nil {
 			adsLog.Warn("Invalid value for V2_REFRESH")
-			periodicRefreshDuration = 0 // this is also he default, but setting it explicitly
 		}
 	}
 	if periodicRefreshDuration > 0 {
-		go periodicRefresh()
+		go out.periodicRefresh()
```

Comment:
Periodicrefreshduration is set to 60s in the beginning of this file. I remember you removing that (and the deleted comment above also states the same). Did you intend to set the global periodicRefreshDuration to 0?